sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -3031,6 +3031,52 @@ class RocksDBSuite extends AlsoTestWithRocksDBFeatures with SharedSparkSession
     }
   }
 
+  testWithChangelogCheckpointingEnabled(
+    "SPARK-51717 - validate that RocksDB file mapping is cleared " +
```

Comment:
Do we verify that the test fails without your change above ?

Reply:
yep, wrote the test first to make sure it fails before fixing. if you comment out my fix, it will fail